### PR TITLE
docs: add skaffold upgrade doc

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

**Comment by jazzdan** [comment]
As before, I'll commit the index change and the resulting `make docs` change in a separate commit.

File: docs/upgrade_skaffold.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+From Skaffold
```
Comment (nicks) [inline docs/upgrade_skaffold.md]: should this be an h1 ?

File: docs/upgrade_skaffold.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+## Before you begin
```
Comment (nicks) [inline docs/upgrade_skaffold.md]: i like the template of having a "Before you begin" section!

File: docs/upgrade_skaffold.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+* [Install Tilt](quickstart.html)
+* If you're new to Tilt try stepping through [a Simple Tiltfile](first_config.html) first.
+
+## Differences between Docker Compose and Skaffold
```
Comment (nicks) [inline docs/upgrade_skaffold.md]: is this supposed to say Tilt and Skaffold?

File: docs/_build/html/_sources/upgrade_skaffold.md.txt [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# From Skaffold
+## Before you begin
+* [Install Tilt](quickstart.html)
+* If you're new to Tilt try stepping through [a Simple Tiltfile](first_config.html) first.
+
+## Differences between Skaffold and Tilt
+* Skaffold streams the logs for services it started. Tilt provides a UI right in your console to view all of your running services and other relevant errors in addition to the log view that Skaffold provides.
```
Comment (nicks) [inline docs/_build/html/_sources/upgrade_skaffold.md.txt]: maybe say something like "We found one combined stream of all services difficult to use & understand for anything beyond the most simple apps." i.e., a justification for why Tilt does it differently

**Pull request merged**